### PR TITLE
Fix text builder

### LINKED ISSUE #42: Text builder is broken
The text builder loses all style and click events (ref: /tpa from module-teleport on the AOF3 NA server). Unsure if this is caused by the mixins in module-chat

### PR DESCRIPTION
**Pull request by Martmists-GH**
closes #42

### CODE REVIEW

**Review by SilverAndro — APPROVED** [review]
Looks all good to me

File: essentials-api/src/main/kotlin/io/github/nyliummc/essentials/api/builders/Text.kt [hunk 1]
```diff
@@ -29,7 +29,7 @@ import net.minecraft.text.ClickEvent
 import net.minecraft.text.HoverEvent
 import net.minecraft.util.Formatting
 import java.util.function.Consumer
-import net.minecraft.text.Text as MCText
+import net.minecraft.text.BaseText as MCText
```
Comment (i509VCB) [inline essentials-api/src/main/kotlin/io/github/nyliummc/essentials/api/builders/Text.kt]: Is there a reason we have to upcast to base text?

Comment (Martmists-GH) [reply essentials-api/src/main/kotlin/io/github/nyliummc/essentials/api/builders/Text.kt]: Text itself doesn't have styled or setStyle, so this is required.

Comment (i509VCB) [reply essentials-api/src/main/kotlin/io/github/nyliummc/essentials/api/builders/Text.kt]: Wouldn't MutableText suffice then?

**Pull request merged**